src/components/Api.js

```diff
@@ -0,0 +1,53 @@
+export class Api {
+    constructor(options) {
+        this._baseUrl = options.baseUrl;
+        this._headers = options.headers;
+    }
+
+    async _request(url, method, requestBody) {
+        return fetch(this._baseUrl + url, {
+            method: method,
+            headers: this._headers,
+            body: JSON.stringify(requestBody)
+          })
+            .then((res) => {
+                if (res.ok) {
+                    return res.json()
+                } else {
+                    throw "requeset gone wrong";
```

Comment:
`                    throw "requeset gone wrong";
`

You need to invoke an error if the response is not `ok` according to the task. It should be done by ` return Promise.reject(`

```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```